doc/docusaurus/docs/delve-deeper/casing-constants.md

```diff
@@ -0,0 +1,125 @@
+---
+sidebar_position: 26
+---
+
+# Case-splitting on constants
+
+:::info
+
+This use of `case` in UPLC was introduced with protocol version 11 and requires
+[Plutus Core version](../essential-concepts/versions#plutus-core-version)
+`1.1.0`.
+
+:::
+
+In UPLC, it is possible to branch on expressions of certain built-in types, like
```

Comment:
We should mention other ways of deconstructing builtin types (e.g., for `bool` one can use `ifThenElse`), and why using the new builtin casing feature is better.